### DISCUSSION TITLE
Audio preprocessing

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#267] [Archived]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 6, 2020, 9:37am]<br/> <br/> Hi<br/> <br/> Does anyone have experience on how much consistent normalized audio have<br/> on the final output? *slash* <br/> What level are you normalizing at? - 6db?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/audio-preprocessing]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [September 6, 2020, 5:44pm]<br/> <br/> You can configure TTS/Tacotron to do preprocessing and normalize audio,<br/> so there is no need to do that yourself. But what is helful is to have<br/> consistent average loudness. I preprocessed our dataset with<br/> [RNNoise](https://github.com/xiph/rnnoise) and<br/> [pyloudnorm](https://github.com/csteinmetz1/pyloudnorm) (latter with an<br/> average loudness of -25dB, at bit lower than usual to avoid clipping<br/> artefacts).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 7, 2020, 6:47am]<br/> <br/> > th an average loudness of -25dB, at bit lower than usual to avoid<br/> > clipping ar<br/> <br/> Thanks. That probably makes it easier. I use desktop software (Audacity)<br/> to process all files right now.<br/> <br/> What about other noise and background rumble in the audio, to what<br/> extent will that affect the final TTS?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [September 7, 2020, 7:02am]<br/> <br/> > What about other noise and background rumble in the audio, to what<br/> > extent will that affect the final TTS?<br/> <br/> Audio signal should be as clean as possible. TTS model can not<br/> distinguish between speech signal and unwanted background noise and will<br/> learn e.g. room echo noise as part of the signal otherwise - which leads<br/> to echoing and/or robotic voice sound.<br/> <br/> As already mentioned RNNoise removes a lot of noise very effectively. In<br/> addition you can do lowpass (e.g. 8000Hz) and highpass (e.g. 50Hz)<br/> filtering and set fmin/fmax parameters in config file.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 9, 2020, 10:44am]<br/> <br/> > As already mentioned RNNoise removes a lot of noise very effectively.<br/> > In addition you can do lowpass (e.g. 8000Hz) and highpass (e.g. 50Hz)<br/> > filtering and set fmin/fmax parameters in config file.<br/> <br/> Thanks a lot. Where can i read more about this? I suppose this is<br/> RNNoise? <https://github.com/xiph/rnnoise><br/> <br/> Are the other variables entered into the config?<br/> <br/> > (e.g. 8000Hz) and highpass (e.g. 50Hz) filtering and set fmin/fmax<br/> <br/> Do they need to be set during the whole training or just used when<br/> syntheseizing? *slash* <br/> Or can i change them later once the model has been trained for a while?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> CrazyJoeDevola<br/> <DATE> [September 9, 2020, 10:46am]<br/> <br/> Also, sometimes i get a long 'noise trail' when rendering audio using<br/> either TTS test audios or together with Pwgan decoder.<br/> <br/> I wanna apply the best possible vocoder (doesnt matter if it takes<br/> longer time). Which would your recommend?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [September 9, 2020, 1:00pm]<br/> <br/> > I suppose this is RNNoise? <https://github.com/xiph/rnnoise><br/> <br/> Yes, that is the correct one.<br/> <br/> > Do they need to be set during the whole training or just used when<br/> > syntheseizing?<br/> <br/> fmin/fmax are parameters for TTS internal audio-processor and must be<br/> set at training time, they don't have effect at inference time. *slash* <br/> As you want to train a vocoder model make sure to use the same audio<br/> process settings as for the training of the Taco/TTS model. If I<br/> remember correctly the dev-branch already uses same config for Taco and<br/> Vocoder training.<br/> <br/> > I wanna apply the best possible vocoder (doesnt matter if it takes<br/> > longer time). Which would your recommend?<br/> <br/> I don't have too much practical experience with vocoders yet, but to my<br/> understanding best results can be achieved with ParallelWaveGAN and/or<br/> Multiband-MelGAN.<br/> <br/> [Archived Post]<br/>

**Comment by taras-sereda** [comment]
Thanks for sharing details on audio normalization and denoising! 
I'm curious if you have any comparison of model performance on loud-normalized audio vs not normalized?

Hypothesis:
pyludnorm uses not very sophisticated loudness normalization algorithm, so a model can potentially learn necessary filters to perform renormalization.

The same question is on the performance comparison of models trained on denoised vs not denoised data.

The statements posted above make sense to me, that training models on normalized and denoised data is easier than on noisy data. Though on the other hand abundance of noisy speech available on the web, makes it very appealing to use it for TTS models.

**Reply by erogol** [reply]
If denoiser is good, then denoised data is always better. 

Loud normalization also helps. I find just min-max norm and * 0.9 works as good as any advance loud-normalization
